src/Silex/Provider/SecurityServiceProvider.php

```diff
@@ -73,9 +76,26 @@ public function register(Application $app)
         $app['security.hide_user_not_found'] = true;
 
         $app['security'] = $app->share(function ($app) {
+            // Deprecated, to be removed in 2.0
             return new SecurityContext($app['security.authentication_manager'], $app['security.access_manager']);
```

Comment:
this won't work in new versions of Symfony: passing the old dependencies will create other instances of the TokenStorage and AuthorizationChecker checker internally. While this is not really an issue for the AuthorizationChecker as it is a stateless service, it is an issue for the TokenStorage, because you would then have 2 different storages, and only one of them would have the right token in it.

For the new interface, the ``security`` service should be defined with its new dependencies